src/gui/plugins/plotting/Plotting.hh

```diff
@@ -0,0 +1,175 @@
+  /// \param[in] _value value to be set to the attribute
+  public: void SetAttributeValue(std::string _attribute, const double &_value);
+
+  /// \brief Get all attributes of the component
+  /// \return component attributes
+  public: std::map<std::string, ignition::gui::PlotData*> Data() const;
+
+  /// \brief Get the Component entity ID
+  /// \return entity ID
+  public: uint64_t Entity();
+
+  /// \brief Get the Component type ID
+  /// \return component type ID
+  public: uint64_t TypeId();
+
+  /// \brief dataPtr holds Abstraction data of PlottingPrivate
+  private: std::unique_ptr<PlotComponentPrivate> dataPtr;
+};
+
+class PlottingPrivate;
+
+class Plotting : public ignition::gazebo::GuiSystem
+{
+  Q_OBJECT
+  /// \brief Constructor
+  public: Plotting();
+
+  /// \brief Destructor
+  public: ~Plotting();
+
+    /// \brief called every simulation iteration to access components
+  public: void Update(const ignition::gazebo::UpdateInfo &_info,
+                      ignition::gazebo::EntityComponentManager &_ecm) override;
+
+  /// \brief Set the Component data of giving id to the giving vector
+  /// \param [in] _Id Component Key of the components map
+  /// \param [in] _vector Vector Data to be set to the component
+  public: void SetData(std::string _Id,
+                       const ignition::math::Vector3d &_vector);
+
+  /// \brief Set the Component data of giving id to the giving pose
+  /// \param [in] _Id Component Key of the components map
+  /// \param [in] _pose Position Data to be set to the component
+  public: void SetData(std::string _Id,
+                       const ignition::math::Pose3d &_pose);
+
+  /// \brief Set the Component data of giving id to the giving number
+  /// \param [in] _Id Component Key of the components map
+  /// \param [in] _value double Data to be set to the component
+  /// valid for types (double, float, int, bool)
+  public: void SetData(std::string _Id, const double &_value);
+
+  /// \brief Add a chart to a specefic component attribute
+  /// \param [in] _entity entity id in the simulation
+  /// \param [in] _typeId type identifier unique to each component type
+  /// \param [in] _attribute component attribute to add the chart to it
+  /// ex: x attribute in Pose3d Component
+  /// \param [in] _chart chart ID to be registered
+  public slots: void RegisterChartToComponent(uint64_t _entity,
+                                              uint64_t _typeId,
+                                              std::string _type,
+                                              std::string _attribute,
+                                              int _chart);
+
+  /// \brief Remove a chart from a specefic component attribute
+  /// \param [in] _entity entity id in the simulation
+  /// \param [in] _typeId type identifier unique to each component type
+  /// \param [in] _attribute component attribute to remove the chart from it
+  /// ex: x attribute in Pose3d Component
```

Comment:
I think it would be helpful if your example also mentioned the string. For the "x attribute in Pose3d Component", what does `_attribute` look like?